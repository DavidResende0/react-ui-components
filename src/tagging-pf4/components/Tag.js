import * as React from 'react';
import PropTypes from 'prop-types';
import { Badge } from '@patternfly/react-core'

import TaggingPropTypes from '../TaggingPropTypes';
import BadgeWithDelete from './BadgeWithDelete'

const tooltip = text => <Tooltip id="tooltip">{text}</Tooltip>;

const Tag = ({
  onTagDeleteClick, tagCategory, tagValue, truncate,
}) => (
  <li key={tagValue.id} className="pf4-tag">
    <BadgeWithDelete
      onDeleteClick={onTagDeleteClick}
      description={tagValue.description}
      truncate={truncate}
    />
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagValue: TaggingPropTypes.value,
  truncate: PropTypes.func.isRequired,
};

export default Tag;
