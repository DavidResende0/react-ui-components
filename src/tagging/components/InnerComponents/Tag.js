import * as React from 'react';
import PropTypes from 'prop-types';
import { Label, OverlayTrigger, Tooltip } from 'patternfly-react';
import TaggingPropTypes from '../TaggingPropTypes';

const tooltip = text => <Tooltip id="tooltip">{text}</Tooltip>;

const Tag = ({
  onTagDeleteClick, tagCategory, tagValue, truncate,
}) => (
  <li key={tagValue.id} className="tag">
    <OverlayTrigger placement="bottom" overlay={tooltip(tagValue.description)}>
      <Label
        key={tagValue.id}
        bsStyle="primary"
        onRemoveClick={() => onTagDeleteClick(tagCategory, tagValue)}
        className="tagColor"
      >
        {truncate(tagValue.description)}
      </Label>
    </OverlayTrigger>
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: TaggingPropTypes.category,
  tagValue: TaggingPropTypes.value,
  truncate: PropTypes.func.isRequired,
};

export default Tag;
