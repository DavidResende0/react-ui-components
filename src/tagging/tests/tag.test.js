import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tag from '../components/tag';

const tagCategory = { description: 'animal', id: 1 };
const tagValue = { description: 'duck', id: 1 };
const onDelete = x => x;

describe('Tag Component', () => {
  it('match snapshot', () => {
    const tree = mount(<Tag tagCategory={tagCategory} tagValue={tagValue} onTagDeleteClick={onDelete} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
