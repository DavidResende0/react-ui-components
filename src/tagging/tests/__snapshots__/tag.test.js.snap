// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component match snapshot 1`] = `
<Tag
  onTagDeleteClick={[Function]}
  tagCategory={
    Object {
      "description": "animal",
      "id": 1,
    }
  }
  tagValue={
    Object {
      "description": "duck",
      "id": 1,
    }
  }
>
<<<<<<< HEAD
  <li
    className="tag"
    key="1"
  >
    <Label
      bsStyle="primary"
      className="tagColor"
      key="1"
      onRemoveClick={[Function]}
      type="default"
=======
  <span
    className="tagColor label label-primary"
    onBlur={[Function]}
    onClick={null}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <a
      className="pf-remove-button"
      href="#"
      onClick={[Function]}
>>>>>>> Change tests
    >
      <Label
        bsClass="label"
        bsStyle="primary"
        className="tagColor"
      >
        <span
          className="tagColor label label-primary"
        >
          duck
          <RemoveButton
            className=""
            onRemoveClick={[Function]}
            title="Remove"
          >
            <a
              className="pf-remove-button"
              href="#"
              onClick={[Function]}
            >
              <span
                aria-hidden="true"
                className="pficon pficon-close"
              />
              <span
                className="sr-only"
              >
                Remove
              </span>
            </a>
          </RemoveButton>
        </span>
      </Label>
    </Label>
  </li>
</Tag>
`;
